### PR TITLE
Convert to use new cli

### LINKED ISSUE #101: Convert to use new CLI Template
Change the current structure of the application to use the new MagicScript CLI template.

### PR DESCRIPTION
**Pull request by panwrona**
Resolve #101

### CODE REVIEW

**Comment by kpiascik** [comment]
Do you need to update the GitHub Actions to work with this?

**Comment by panwrona** [comment]
> Do you need to update the GitHub Actions to work with this?

@kpiascik No, I updated GH Actions in previous commits. But I need to check if they're working properly

**Comment by panwrona** [comment]
@darek607 @grzegorzdec @SPaluchiewiczSofomo @pleszkiewicz If you have time, I encourage you guys to checkout to this branch and test new Catalog structure

**Comment by SPaluchiewiczSofomo** [comment]
@panwrona my findings from test - I think they're more related to `magic-script-cli` or new app template but it's worth to put it here:
- `magic-script run ios` does nothing and `magic-script install ios` too (maybe there should be info that user should use `build`)
- when you run `magic-script build ios` after `magic-script build` (implicit for Lumin) there is `node_module` conflict

File: LICENSE [hunk 1]
```diff
@@ -1,201 +0,0 @@
-                                 Apache License
```
Comment (kpiascik) [inline LICENSE]: do not remove this file

File: NOTICE [hunk 2]
```diff
@@ -1,6255 +0,0 @@
-   Copyright 2019 Magic Leap, Inc.
```
Comment (kpiascik) [inline NOTICE]: Do not remove this file.

File: README.md [hunk 3]
```diff
@@ -1,65 +0,0 @@
-# ARDemo project
```
Comment (kpiascik) [inline README.md]: don't remove this file either, but simple put the steps to build and run using the CLI

File: lumin/app.package [hunk 4]
```diff
@@ -1,4 +1,4 @@
 DATAS = "digest.sha512.signed" : "." \
-        "bin/" : "bin/" \
-        "resources/" : "resources/"
+        "bin/" : "bin/"\
+        "../resources/" : "resources/"
```
Comment (kpiascik) [inline lumin/app.package]: don't we also want to include the `assets` folder?

File: lumin/package.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    "rollup": "^1.1.2",
+    "rollup-plugin-babel": "^4.3.2",
+    "rollup-plugin-commonjs": "^9.3.4",
+    "rollup-plugin-includepaths": "0.2.3",
+    "rollup-plugin-node-resolve": "^4.0.0",
+    "semistandard": "^13.0.1"
+  },
+  "dependencies": {
+    "magic-script-components-lumin": "1.0.13",
+    "magic-script-polyfills": "^2.4.2"
```
Comment (kpiascik) [inline lumin/package.json]: this should be 2.7.1

File: lumin/rollup.config.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  // Build for MagicScript on Magicverse (iOS, Android)
+  {
+    ...common,
+    input: '../src/app.js',
+    external: ['react'],
+    output: {
+      globals: {
+        'react': 'React'
+      },
+      file: '../reactnative/bin/bundle.js',
```
Comment (kpiascik) [inline lumin/rollup.config.js]: this should go into `lumin/bin` not reactnative.  This needs to be fixed in the template as well

File: reactnative/assets [hunk 7]
```diff
@@ -0,0 +1 @@
+../assets
```
Comment (kpiascik) [inline reactnative/assets]: isn't this a symlink?  should it be committed?

File: reactnative/ios/Catalog/Info.plist [hunk 8]
```diff
@@ -37,10 +37,8 @@
 			</dict>
 		</dict>
 	</dict>
-	<key>NSCameraUsageDescription</key>
```
Comment (kpiascik) [inline reactnative/ios/Catalog/Info.plist]: we need the camera usage description and the text cannot be empty

File: reactnative/ios/Catalog/Info.plist [hunk 9]
```diff
@@ -37,10 +37,8 @@
...
-	<string>Application needs access to the camera in order to display AR world correctly.</string>
 	<key>NSLocationWhenInUseUsageDescription</key>
-	<string>Application needs access to Location Services in order to track your location properly.</string>
+	<string></string>
```
Comment (kpiascik) [inline reactnative/ios/Catalog/Info.plist]: same applies to the Location

**Comment by panwrona** [comment]
@kpiascik 
We think that catalog is ready, but there are some more issues: 
a) PlatformOS is not released in magic-script-components yet, but it’s necessary to choose if we want display animated model on android or not) 
b) On iOS dropdown list between prev and next button is not scrollable
 c) When Circle Confirmation finishes it crashes Lumin 
d) Lumin doesn’t provide ability to download remote sounds

File: reactnative/package.json [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    "@babel/plugin-transform-react-jsx": "^7.3.0",
+    "@babel/plugin-proposal-class-properties": "^7.4.4",
+    "babel-jest": "^24.8.0",
+    "jest": "^24.8.0",
+    "metro-react-native-babel-preset": "^0.54.1",
+    "react-test-renderer": "16.8.3",
+    "magic-script-typings": "1.4.0"
+  },
+  "dependencies": {
+    "magic-script-polyfills": "2.4.2",
```
Comment (kpiascik) [inline reactnative/package.json]: we don't need polyfills for react-native do w?

File: reactnative/package.json [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+  "devDependencies": {
+    "@babel/core": "^7.4.5",
+    "@babel/runtime": "^7.4.5",
+    "@babel/plugin-transform-react-jsx": "^7.3.0",
+    "@babel/plugin-proposal-class-properties": "^7.4.4",
+    "babel-jest": "^24.8.0",
+    "jest": "^24.8.0",
+    "metro-react-native-babel-preset": "^0.54.1",
+    "react-test-renderer": "16.8.3",
+    "magic-script-typings": "1.4.0"
```
Comment (kpiascik) [inline reactnative/package.json]: we don't need the typings here either

File: reactnative/resources [hunk 12]
```diff
@@ -0,0 +1 @@
+../resources
```
Comment (kpiascik) [inline reactnative/resources]: same question as for the `assets` folder.  Isn't this a symlink? and if so should it be committed?

File: reactnative/assets (hunk 7 above)
Comment (panwrona) [reply reactnative/assets]: Yes, it's a symlink and it should be committed in order to work on React Native.

File: reactnative/resources (hunk 12 above)
Comment (panwrona) [reply reactnative/resources]: Yes, it's a symlink and it should be committed in order to work on React Native.

File: lumin/app.package (hunk 4 above)
Comment (panwrona) [reply lumin/app.package]: Done

File: lumin/package.json (hunk 5 above)
Comment (panwrona) [reply lumin/package.json]: Done

File: lumin/rollup.config.js (hunk 6 above)
Comment (panwrona) [reply lumin/rollup.config.js]: But do we really need it here? We don't have any Magicverse dependencies in this project yet

File: clean_hard.sh [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+rm -fr lumin/package-lock.json
+
+echo "3. cleaning reactnative folder..."
+rm -fr reactnative/ios/Pods
+rm -fr reactnative/ios/Podfile.lock
+rm -fr reactnative/ios/*.xcworkspace
+rm -fr reactnative/node_modules
+rm -fr reactnative/package-lock.json
+
+say -v Melina "clean done"
```
Comment (kpiascik) [inline clean_hard.sh]: what does this do?

File: lumin/rollup.config.js (hunk 6 above)
Comment (kpiascik) [reply lumin/rollup.config.js]: Please update this.  It will make our lives easier having it in place correctly.

File: clean_hard.sh (hunk 13 above)
Comment (SPaluchiewiczSofomo) [reply clean_hard.sh]: It cleans `npm install` results and when it's done says (via audio output device) `clean done` (with Melina lovely voice).
It's very useful as you can switch to different task and switch back when you hear this `voice notification`.

Comment (kpiascik) [reply clean_hard.sh]: I was specifically asking about the last line. The say command

Comment (panwrona) [reply clean_hard.sh]: It just says 'clean done' on the mac machine

**Pull request merged**